codegen/src/lib.rs

```diff
@@ -1,310 +1,96 @@
+        "Couldn't open file for writing: {:?}",
+        output_path,
+    ));
 
-// Translate botocore "shapes" to Rust types
-fn render_shapes(service: &Service) -> String {
-    let mut src = String::new();
-    for (name, shape) in service.shapes.iter() {
-        // String is already a type in Rust
-        if name == "String" {
-            continue;
-        }
-        if shape.shape_type == "structure" {
-            src = src + &format!("{}\n", struct_type(name, &shape));
-        } else {
-            let rust_type = match &*shape.shape_type {
-                "structure" => struct_type(name, &shape),
-                "map" => format!("::std::collections::HashMap<{},{}>", shape.key(), shape.value()),
-                "list" => format!("Vec<{}>", shape.member()),
-                _ => primitive_type(&shape.shape_type)
-            };
-            src = src + &format!("pub type {} = {};\n", name, rust_type);
-        }
-    }
-    src
+    output_file.write_all(source_code.as_bytes()).expect(&format!(
+        "Failed to write generated source code to {:?}",
+        output_path,
+    ));
 }
 
+#[cfg(not(feature = "serde_macros"))]
+fn serde_generate(source: &Path, destination: &Path) {
+    let mut registry = ::syntex::Registry::new();
 
-fn struct_type(name: &str, shape: &Shape) -> String {
-    if shape.members.is_empty() {
-        return format!("#[derive(Debug, Serialize, Deserialize, Default)]\npub struct {};", name);
-    }
-    let mut struct_type = format!("#[derive(Debug, Serialize, Deserialize, Default)]\npub struct {} {{\n", name);
-    for (member_name, member) in shape.members.iter() {
-        if member.documentation.is_some() {
-            //struct_type = struct_type + "\t// documentation\n";
-        }
-        if shape.required(member_name) {
-            struct_type = struct_type + &format!("\tpub {}: {},\n", member_name, member.shape);
-        } else {
-            // There's surely a better way to do this:
-            if member_name == "type" {
-                struct_type = struct_type + &format!("\tpub aws_{}: Option<{}>,\n", member_name, member.shape)
-            }
-            else {
-                struct_type = struct_type + &format!("\tpub {}: Option<{}>,\n", member_name, member.shape)
-            }
-        }
-    }
-    struct_type = struct_type + "}\n";
-    struct_type
-}
-
-fn primitive_type(shape_type: &str) -> String {
-    match shape_type {
-        "string" => "String".to_string(),
-        "integer" => "i32".to_string(),
-        "long" => "i64".to_string(),
-        "float" => "f32".to_string(),
-        "double" => "f64".to_string(),
-        "blob" => "Vec<u8>".to_string(),
-        "boolean" => "bool".to_string(),
-        // yes, this is a float type for a timestmap.
-        // that's how it comes back from AWS
-        "timestamp" => "f64".to_string(),
-        _ => panic!(format!("Unknown type '{}'", shape_type))
-    }
+    ::serde_codegen::register(&mut registry);
```

Comment:
From the Rust reference:

> Paths starting with `::` are considered to be global paths where the components of the path start being resolved from the crate root. Each identifier in the path must resolve to an item.

It lets you reference something by its fully-qualified path without importing it into the module.